mypy/checkexpr.py

```diff
@@ -2981,8 +2981,8 @@ def visit_index_with_type(self, left_type: Type, e: IndexExpr,
                 return make_simplified_union(out)
             else:
                 return self.nonliteral_tuple_index_helper(left_type, index)
-        elif isinstance(left_type, TypedDictType):
-            return self.visit_typeddict_index_expr(left_type, e.index)
+        # elif isinstance(left_type, TypedDictType):
```

Comment:
Related: https://github.com/python/mypy/pull/11244

Reply:
Yeah it's a rollback. Do you have opinions about the original proposal https://github.com/python/mypy/issues/6138? It that case maybe we should alse move delitem and setitem to plugins